File: mobile/ios/OmenIOS/OmenIOS/DesignSystem/OmenConfirmationDialog.swift
import SwiftUI

/// Registry §3.1 ConfirmationDialog variant. Destructive routes the confirm button through
/// the SwiftUI `.destructive` role so the system renders it as danger; the label still
/// carries the meaning (registry §4; facts-of-record #7).
enum OmenConfirmationVariant { case `default`, destructive }

/// SwiftUI matches the registry §3.1 spec: `.confirmationDialog(...)` is the platform
/// primitive. This modifier keeps every caller on the same tokens/roles.
///
/// Not a login/OTP flow, not a multi-option sheet — for those, see [OmenModalSheet]. Cancel
/// gets an implicit `.cancel`-role button so system a11y and outside-touch dismissal work
/// unchanged.
extension View {
    func omenConfirmationDialog(
        title: String,
        message: String,
        isPresented: Binding<Bool>,
        confirmLabel: String,
        cancelLabel: String,
        variant: OmenConfirmationVariant = .default,
        onConfirm: @escaping () -> Void
    ) -> some View {
        confirmationDialog(
            title,
            isPresented: isPresented,
            titleVisibility: .visible
        ) {
            switch variant {
            case .default:
                Button(confirmLabel, action: onConfirm)
            case .destructive:
                Button(confirmLabel, role: .destructive, action: onConfirm)
            }
            Button(cancelLabel, role: .cancel) {}
        } message: {
            Text(message)
        }
    }
}

File: mobile/android/core/designsystem/src/main/kotlin/com/slopssaloon/omen/core/designsystem/component/OmenConfirmationDialog.kt
package com.slopssaloon.omen.core.designsystem.component

import androidx.compose.material3.AlertDialog
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.slopssaloon.omen.core.designsystem.theme.OmenTheme

/**
 * Registry §3.1 ConfirmationDialog: presents a titled decision with a required confirm
 * action and a cancel escape hatch. Destructive variant uses `data.riskHigh` for the confirm
 * button so the danger is a *redundant* signal — the button label still carries the meaning
 * (registry §4; facts-of-record #7).
 *
 * Not a login/OTP flow, not a bottom sheet with multiple options — for those, see
 * [OmenModalSheet]. This wraps Material 3 [AlertDialog] so system a11y, Dynamic Type,
 * outside-touch dismissal, and back-button handling are inherited.
 */
enum class OmenConfirmationVariant { Default, Destructive }

@Composable
fun OmenConfirmationDialog(
    visible: Boolean,
    title: String,
    message: String,
    confirmLabel: String,
    cancelLabel: String,
    onConfirm: () -> Unit,
    onDismiss: () -> Unit,
    variant: OmenConfirmationVariant = OmenConfirmationVariant.Default,
    modifier: Modifier = Modifier,
) {
    if (!visible) return
    val colors = OmenTheme.color
    val confirmColor = when (variant) {
        OmenConfirmationVariant.Default -> colors.accent
        OmenConfirmationVariant.Destructive -> colors.data.riskHigh
    }

    AlertDialog(
        onDismissRequest = onDismiss,
        modifier = modifier,
        containerColor = colors.surface1,
        titleContentColor = colors.textPrimary,
        textContentColor = colors.textSecondary,
        title = { Text(title, style = OmenTheme.typography.h2.toTextStyle()) },
        text = { Text(message, style = OmenTheme.typography.body.toTextStyle()) },
        confirmButton = {
            TextButton(
                onClick = onConfirm,
                colors = ButtonDefaults.textButtonColors(contentColor = confirmColor),
            ) { Text(confirmLabel, style = OmenTheme.typography.body.toTextStyle()) }
        },
        dismissButton = {
            TextButton(
                onClick = onDismiss,
                colors = ButtonDefaults.textButtonColors(contentColor = colors.textSecondary),
            ) { Text(cancelLabel, style = OmenTheme.typography.body.toTextStyle()) }
        },
    )
}
